### ISSUE TITLE
bug(TextField): Password text field has no focus indicator for eye icon button

### ISSUE THREAD

**Issue by ling1726** [microsoft/fluentui#18900]
### Environment Information

- **Package version(s)**: ^8.0.0
- **Browser and OS versions**: Chromium edge/Chrome

### Describe the issue:

TextField password variant does not have a focus indicator on the eye icon

### Please provide a reproduction of the issue in a codepen:

[Repro on official docsite](https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield)

#### Actual behavior:

Tab from the password input actually focuses the eye icon button to hide reveal password.

![image](https://user-images.githubusercontent.com/20744592/125279568-63e90a00-e314-11eb-970b-020b7943f46a.png)

No focus indicator when the eye icon is focused

#### Expected behavior:

Focus indicator when the eye icon is focused

**Comment by ling1726** [comment]
@behowell please find a home for this in the v8 team

@theboyknowsclass Created this issue from #18864
